### PR TITLE
Implements ability to resolve same named constraints and includes CLEAR keywords

### PR DESCRIPTION
**Pull request by jonnyh404**

### CODE REVIEW

**Review by kwr3 — APPROVED** [review]
looks good

File: ResSimpy/DataModelBaseClasses/OperationsMixin.py [hunk 1]
```diff
@@ -166,3 +167,46 @@ def resolve_same_named_objects(sorted_by_date: Sequence[T]) -> Sequence[T]:
                     setattr(new_resolved_object, attr, value)
             resolved_objects.append(new_resolved_object)
         return resolved_objects
+
+    @staticmethod
+    def resolve_same_named_objects_constraints(sorted_by_data: Sequence[NexusConstraint]) -> Sequence[NexusConstraint]:
+        """Resolves a subset of objects by date and applies clears in the constraints."""
```
Comment (cflynn3) [inline ResSimpy/DataModelBaseClasses/OperationsMixin.py:173]: Please add docstrings

File: tests/test_operations_mixin.py [hunk 2]
```diff
@@ -109,3 +110,52 @@ def test_resolve_carried_over_attributes_same_date():
...
+                        max_pressure=100),
+        NexusConstraint(max_reservoir_oil_rate=120, name='well1', date='01/03/2020', date_format=DateFormat.DD_MM_YYYY,
+                        max_pressure=100),
+    ]
+    
+    # Act
+    result = NetworkOperationsMixIn.resolve_same_named_objects_constraints(constraints_for_well)
+    
+    # Assert
+    assert result == expected_result
```
Comment (cflynn3) [inline tests/test_operations_mixin.py:137]: Please add other test cases for clearing just some of the others e.g. pressure.

**Review by cflynn3 — APPROVED** [review]
Looks good, please look at my comments.

**Pull request merged**